Realm/RLMRealm.mm

```diff
@@ -487,10 +487,21 @@ - (void)removeNotification:(RLMNotificationToken *)token {
 - (void)sendNotifications:(NSString *)notification {
     NSAssert(!_realm->config().read_only, @"Read-only realms do not have notifications");
 
+    NSUInteger count = _notificationHandlers.count;
```

Comment:
`_notificationHandlers.allObjects` is much slower than `anyObject`, so this avoids calling it unless it's actually needed.